.github/actions/setup-rust/action.yml

```diff
@@ -8,19 +8,14 @@ runs:
       shell: bash
       run: echo "version=$(cat rust-toolchain.toml | grep channel | awk -F'\"' '{print $2}')" >> $GITHUB_OUTPUT
 
-    - name: Rust Toolchain Cache
-      id: rustup-cache
-      uses: actions/cache@v4
-      with:
-        path: ~/.rustup
-        key: "rustup-${{ runner.os }}-${{ steps.rust-version.outputs.version }}"
```

Comment:
not sure how to make this work without reimplementing rust-toolchain action so I axed it. We can build our own rust-toolchain action if necessary that tries to rigure out caching. Sadly the rust version resolution seems to use rustc but maybe it's possible to do with just rustup